DataProcessing/datax-host/src/main/scala/datax/host/BlobBatchingHost.scala

```diff
@@ -102,5 +101,38 @@ object BlobBatchingHost {
 
     appLog.warn(s"Batch Mode Work Ended, processed metrics: $batchResult")
     AppInsightLogger.trackEvent(ProductConstant.ProductRoot + "/batch/end", null, batchResult)
+
+    //write tracker file
+    val (needWriteTracker, trackerFolder, dateTime) = getTrackerConfigs(inputArguments)
+    if (needWriteTracker) {
+      writeTracker(trackerFolder, dateTime)
+    }
+  }
+
+  // get tracker configs from input arguments and sys env
+  // if the trackerFolder is configured, a tracker file should be written in the folder
+  def getTrackerConfigs(inputArguments: Array[String]) : (Boolean, String, Timestamp) = {
+    val namedArgs = ArgumentsParser.getNamedArgs(inputArguments)
```

Comment:
This is already available as part of conf in the initApp method in CommonAppHost.scala .. line "val conf = ConfigManager.getConfigurationFromArguments(inputArguments)"

Reply:
The idea is to read the config only once at the start of the job and passing along the parameters value wherever required